src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,615 @@
+    EMOTE_EMERGE_SOON            = 2,
+};
+
+enum Says
+{
+    SAY_PLAYER_TEXT_1            = 0,
+    SAY_PLAYER_TEXT_2            = 1,
+    SAY_PLAYER_TEXT_3            = 2
+};
+
+enum Events
+{
+    EVENT_EMERGE                 = 1,
+    EVENT_SUBMERGE               = 2,
+    EVENT_ICE_SPEAR              = 3,
+    EVENT_COLD_SLAP              = 4,
+    EVENT_FLAMECALLER_BEAM       = 5,
+    EVENT_SUMMON_HAILSTONE       = 6,
+    EVENT_SUMMON_COLDWEAVE       = 7,
+    EVENT_SUMMON_FROSTWIND       = 8,
+    EVENT_SUMMON_SPEAR_BUNNY     = 9,
+    EVENT_EMOTE_EMERGE           = 10,
+    EVENT_EMOTE_SUBMERGE         = 11,
+    EVENT_EMOTE_EARTHEN_ATTACK   = 12,
+    EVENT_GHOST_VISUAL           = 13,
+    EVENT_REMOVE_GHOST_VISUAL    = 14,
+    EVENT_RESURFACE_SOON         = 15,
+    EVENT_EARTHEN_RING_ATTACK    = 16,
+    EVENT_ER_ATTACK_A            = 17,
+    EVENT_ER_ATTACK_B            = 18,
+    EVENT_ER_ATTACK_C            = 19
+};
+
+enum Actions
+{
+    ACTION_START_EVENT           = -2574500,
+    ACTION_AHUNE_RETREAT         = -2586500
+};
+
+enum Phases
+{
+    PHASE_ONE       = 0,
+    PHASE_TWO       = 1
+};
+
+Position const SummonPositions[] =
+{
+    { -95.523705f, -230.299057f, -1.256438f, 1.637029f },  // Ahune / Frozen-Core
+    { -135.871445f, -145.185211f, -1.851782f, 5.427049f }, // Totem A
+    { -129.179459f, -141.099503f, -1.992221f, 5.160010f }, // Totem B
+    { -122.147339f, -136.741470f, -2.285978f, 5.132535f }, // Totem C	
+    { -140.734543f, -139.580399f, -1.591881f, 5.427049f }, // Flamecaller A
+    { -134.587418f, -134.331085f, -1.855661f, 5.336747f }, // Flamecaller B
+    { -128.186356f, -130.267303f, -2.016853f, 5.460842f }, // Flamecaller C
+    { -73.4977f, -155.976f, -1.8987f, 3.23816f }           // Slippery Floor
+
+};
+
+class boss_ahune : public CreatureScript
+{
+public:
+    boss_ahune() : CreatureScript("boss_ahune") { }
+
+    struct boss_ahuneAI : public BossAI
+    {
+        boss_ahuneAI(Creature* creature) : BossAI(creature, NPC_AHUNE), _summons(me) { }
+
+        InstanceScript* instance = me->GetInstanceScript();
+        uint64 _attacks;
+        EventMap _events;
+        bool _submerged;
+        SummonList _summons;
+    
+        void Initialize()
+        {
+            _attacks = 0;
+            _events.Reset();
+            _summons.DespawnAll();
+            _submerged = false;
+            me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_KNOCK_BACK, true);
+            me->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_KNOCK_BACK_DEST, true);
+            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE);
+            me->CastSpell(me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), SPELL_SUMMONING_VISUAL1, true);
+            me->AddAura(SPELL_AHUNES_SHIELD, me);
+            me->SummonCreature(NPC_FROZEN_CORE, SummonPositions[0], TEMPSUMMON_CORPSE_DESPAWN);
+        }
+
+        void Reset() override
+        {
+            Initialize();
```

Comment:
use _Reset() method and remove _events.Reset(); and _summons.DespawnAll(); from Initialize() and also put Initialize inside constructor as well, its the reason why we use this method